src/loaders/compute_tools/gtdb_tk/gtdb_tk.py

```diff
@@ -73,14 +73,20 @@ def _run_gtdb_tk(ids_to_files: Dict[Path, str], output_dir: Path, threads: int,
 
         for genome_id, classify_res in zip(summary_df[loader_common_names.GTDB_GENOME_ID_COL],
                                            summary_df[loader_common_names.GTDB_CLASSIFICATION_COL]):
+            genome_ids.add(genome_id)
             if classify_res.startswith(loader_common_names.GTDB_UNCLASSIFIED):
-                kbase_id = meta_dict[genome_id]
-                source_file_path = ids_to_files[genome_id]
                 error_message = f"GTDB_tk classification failed: {classify_res}"
-                fatal_tuples.append(FatalTuple(kbase_id, error_message, str(source_file_path), None))
+                fatal_tuple = create_gtdbtk_fatal_tuple(genome_id, meta_dict, ids_to_files, error_message, None)
+                fatal_tuples.append(fatal_tuple)
     
+    miss_genome_ids = set(meta_dict.keys()) - genome_ids
+    for miss_genome_id in miss_genome_ids:
+        error_message = f"Missing classification attribute from {miss_genome_id}"
```

Comment:
From reading the code it seems like there was no result for the genome at all, not just the classification attribute.

Is there anything in the GDTB_tk logs about missing data?

Reply:
Right, there are basically 2 records that have no result at all from gtdb_tk tool.  

From what I see, I did not find anything in the GDTB_tk logs related to missing data.

Reply:
That seems very weird. Are there any docs about whether GTDB_tk should ever silently ignore genomes? I'm wondering if there's a bug in their or our code.

Reply:
It seems ppl had similar issues before with GTDB_tk: https://github.com/Ecogenomics/GTDBTk/issues/356

I `cat` every single file in the batch_dir and found these 2 records in `align/gtdbtk.ar53.filtered.tsv`. It seems they get filtered out because they are not identified as bacterial? GTDB_tk should have put them into *warnings.log or  *failed.genomes.tsv files, which makes more sense to me.

Reply:
Can you find any documentation about these files describing their expected contents?

Reply:
https://ecogenomics.github.io/GTDBTk/files/filtered.tsv.html#files-filtered-tsv

I am also looking at list of args [here](https://ecogenomics.github.io/GTDBTk/commands/classify_wf.html#commands-classify-wf) that can stop excluding genomes.

Reply:
Ok, it looks to me that we should get the genome IDs from `filtered.tsv` and `failed.genomes.tsv` (it'd be nice if they were both in the same file) and throw an error for any missing IDs that aren't in either of those lists

Reply:
Updated!

Reply:
Ah, all our gtdbtk.failed_genomes.tsv files are empty. I am not sure how gtdbtk.failed_genomes.tsv is formatted exactly. But based on their [doc](https://ecogenomics.github.io/GTDBTk/files/failed_genomes.tsv.html), its first col is not genome_id though. 

Have you seen any gtdbtk.failed_genomes.tsv with contents? @Tianhao-Gu

Reply:
There's a couple things you could try:
1. Make a small test set containing some bad data (in `sourcedata/test` ?). Copy in a few good genomes, a couple of the filtered genomes, and copy in a couple of other genomes and edit the fasta file to truncate it so the sequences are short. That might prevent prodigal from calling genes. This is probably a helpful thing to do anyway for testing failure conditions.
2. Read the prodigal docs and see if they mention under what conditions it won't call genes, and see if you can create a fasta file with those conditions.
3. Ask in the #subject-matter-experts channel to see if anyone knows what would cause prodigal / gtdb_tk to fail and if they have any failing fasta files we could use

Reply:
I took the first approach and verified gtdbtk.failed_genomes.tsv's first col is indeed genome_id. 

<img width="711" alt="Screen Shot 2023-06-22 at 2 51 06 PM" src="https://github.com/kbase/collections/assets/22202096/18ab5626-72f2-4872-9972-6073f16d0924">

Reply:
Updated!

Reply:
:+1: